### PR TITLE
Feat/rainbowkit integration

### PR DESCRIPTION
**Pull request by LeifuChen**
#### Wallet
- [x] Connect/Disconnect the Wallet
- [X] Apply the dark/light theme 
- [x] Switch network correctly
- [x] Show ENS name regardless of network
- [x] Show sUSD balance 
- [x] Use `OpenChainModal` to replace the current `NetworkSwitcher`
- [x] Show network name
- [x] Show `futuresPositions` on `BalanceActions`
- [x] Add `Ledger Live` support
- [x] Add `Genosis Safe` support

#### Future
- [x] Deposit & Withdraw margin
- [x] Open/Modify/Close position

#### Exchange
- [x] Show balance on `CurrencyCard` and `selectCurrencyModal`
- [x] Swap synths & erc20 token

## Related issue
#948

### CODE REVIEW

**Comment by Tburm** [comment]
@LeifuChen  Was able to connect using Ledger Live

File: package.json [hunk 1]
```diff
@@ -76,6 +79,9 @@
 		"slick-carousel": "1.8.1",
 		"styled-components": "5.1.1",
 		"styled-media-query": "2.1.2",
+		"synthetix": "2.74.1",
```
Comment (Tburm) [inline package.json:82]: @koredefashokun wasn't this just removed with the decentralized deployment PR?

Comment (LeifuChen) [reply package.json]: Thanks for pointing it out. I will remove the package.

File: sections/shared/Layout/AppLayout/Header/BalanceActions.tsx [hunk 2]
```diff
@@ -31,8 +33,9 @@ const BalanceActions: FC = () => {
 	useQueryCrossMarginAccount();
 
 	const synthBalances = useRecoilValue(balancesState);
+	useGetFuturesMarkets();
+	useGetFuturesPositionForMarkets();
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply sections/shared/Layout/AppLayout/Header/BalanceActions.tsx]: Yes. This is no need to have them any longer as it is invoked in `RefetchContext`.

File: queries/futures/useGetCurrentPortfolioValue.ts [hunk 3]
```diff
@@ -17,20 +17,28 @@ import { mapFuturesPosition } from './utils';
...
 
 	const query = useGetCrossMarginAccountOverview();
 	const freeMargin = query.data?.freeMargin || zeroBN;
 
-	const { synthetixjs } = Connector.useContainer();
+	const synthetixjs = synthetix({
+		provider: getDefaultProvider((activeChain?.id ?? chain.optimism.id) as NetworkId),
+		networkId: (activeChain?.id ?? chain.optimism.id) as NetworkId,
+		useOvm: true,
+	});
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply queries/futures/useGetCurrentPortfolioValue.ts:37]: Yes. Agree that we have to clean this. We got this from `Connector.tsx` before and I think the file will be deprecated.

File: queries/futures/useGetFuturesMarkets.ts [hunk 4]
```diff
@@ -1,35 +1,49 @@
...
 	const isAppReady = useRecoilValue(appReadyState);
-	const network = useRecoilValue(networkState);
-	const { synthetixjs: snxjs, defaultSynthetixjs } = Connector.useContainer();
+	const { chain: activeChain } = useNetwork();
+	const isL2 =
+		activeChain !== undefined
+			? [chain.optimism.id, chain.optimismKovan.id].includes(activeChain?.id)
+			: false;
+	const snxjs = synthetix({
+		provider: getDefaultProvider((activeChain?.id ?? chain.optimism.id) as NetworkId),
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply queries/futures/useGetFuturesMarkets.ts:28]: I remembered the type of provider is different but I will have another try.

File: queries/futures/useGetFuturesPositionForAccount.ts [hunk 5]
```diff
@@ -15,7 +16,11 @@ import { getFuturesEndpoint, mapTradeHistory } from './utils';
 const useGetFuturesPositionForAccount = (options?: UseQueryOptions<any>) => {
 	const { selectedFuturesAddress } = useRecoilValue(futuresAccountState);
 	const isAppReady = useRecoilValue(appReadyState);
-	const isL2 = useRecoilValue(isL2State);
+	const { chain: activeChain } = useNetwork();
+	const isL2 =
+		activeChain !== undefined
+			? [chain.optimism.id, chain.optimismKovan.id].includes(activeChain?.id)
+			: false;
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply queries/futures/useGetFuturesPositionForAccount.ts:23]: Agree that we need to store this in the state.

File: queries/futures/useGetFuturesMarkets.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (LeifuChen) [reply queries/futures/useGetFuturesMarkets.ts]: Yes. We can use the `wagmi` hook here.

File: sections/dashboard/PortfolioChart/PortfolioChart.tsx [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
-import { walletAddressState } from 'store/wallet';
 import { zeroBN } from 'utils/formatters/number';
 
 const PortfolioChart: FC = () => {
 	const portfolioValueQuery = useGetCurrentPortfolioValue();
 	const portfolioValue = portfolioValueQuery?.data ?? null;
+	const { address } = useAccount();
+	const walletAddress = address || null;
 
-	const walletAddress = useRecoilValue(walletAddressState);
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply sections/dashboard/PortfolioChart/PortfolioChart.tsx:16]: Yes. Removed.

File: sections/dashboard/SpotMarketsTable/SpotMarketsTable.tsx [hunk 7]
```diff
@@ -38,7 +36,7 @@ const SpotMarketsTable: FC<SpotMarketsTableProps> = ({ exchangeRates }) => {
 
 	const queryCache = useQueryClient().getQueryCache();
 	// KM-NOTE: come back and delete
-	const frozenSynthsQuery = queryCache.find(['synths', 'frozenSynths', network.id]);
+	const frozenSynthsQuery = queryCache.find(['synths', 'frozenSynths', 10]);
```
Comment (avclarke) [inline sections/dashboard/SpotMarketsTable/SpotMarketsTable.tsx:39]: is this meant to be hardcoded 10?

File: sections/shared/components/GasPriceSelect/GasPriceSelect.tsx [hunk 8]
```diff
@@ -22,8 +24,8 @@ const GasPriceSelect: FC<GasPriceSelectProps> = ({ gasPrices, transactionFee, ..
 	const { t } = useTranslation();
 	const gasSpeed = useRecoilValue(gasSpeedState);
 	const customGasPrice = useRecoilValue(customGasPriceState);
-	const isMainnet = useRecoilValue(isMainnetState);
-	const isL2 = useRecoilValue(isL2State);
+	const isL2 = useIsL2();
+	const isMainnet = useIsL1();
```
Comment (avclarke) [inline sections/shared/components/GasPriceSelect/GasPriceSelect.tsx:28]: Is this correct as mainnet can be l2's and useIsL1 returns Goerli testnet or Eth mainnet it seems?

File: sections/exchange/FooterCard/TradeSummaryCard/GasPriceSummaryItem.tsx [hunk 9]
```diff
@@ -31,8 +31,10 @@ const GasPriceSummaryItem: FC<GasPriceSummaryItemProps> = ({
 }) => {
 	const { t } = useTranslation();
 	const { selectedPriceCurrency } = useSelectedPriceCurrency();
-	const isL2 = useRecoilValue(isL2State);
-	const isMainnet = useRecoilValue(isMainnetState);
+	const { chain: network } = useNetwork();
+	const isL2 = useIsL2();
+	const isMainnet =
+		network !== undefined ? [chain.mainnet.id, chain.goerli.id].includes(network?.id) : false;
```
Comment (avclarke) [inline sections/exchange/FooterCard/TradeSummaryCard/GasPriceSummaryItem.tsx:37]: is this meant to be isL1 as goerli is a testnet and if so could use the useIsL1 hook as it's the same check?

**Review by avclarke — COMMENTED** [review]
Looks great, was taking a look to brace myself for the eventual cross margin merge! Just a couple of minor things.

File: containers/Connector/Connector.tsx [hunk 10]
```diff
@@ -1,291 +1,41 @@
...
-	] = useState<TransactionNotifierInterface | null>(null);
-
+	const { chain: activeChain } = useNetwork();
+	const network =
+		activeChain !== undefined && activeChain.unsupported
+			? chain.optimism
+			: activeChain ?? chain.optimism;
+
+	const provider = useProvider({ chainId: network.id });
+	const L2Provider = useProvider({ chainId: chain.optimism.id });
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply containers/Connector/Connector.tsx:16]: Thanks. Updated.

File: containers/TransactionNotifier/TransactionNotifier.tsx [hunk 11]
```diff
@@ -1,18 +1,25 @@
...
+import { useProvider } from 'wagmi';
 
 import {
 	NotificationSuccess,
 	NotificationPending,
 	NotificationError,
 } from 'components/TransactionNotification';
 import BlockExplorer from 'containers/BlockExplorer';
-import Connector from 'containers/Connector';
+// import Connector from 'containers/Connector';
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply containers/TransactionNotifier/TransactionNotifier.tsx:16]: Removed.

File: queries/futures/useGetAllFuturesTradesForAccount.ts [hunk 12]
```diff
@@ -14,14 +13,17 @@ const useGetAllFuturesTradesForAccount = (
...
-	const isAppReady = useRecoilValue(appReadyState);
-	const network = useRecoilValue(networkState);
-	const futuresEndpoint = getFuturesEndpoint(network);
-	const isWalletConnected = useRecoilValue(isWalletConnectedState);
-	const isL2 = useRecoilValue(isL2State);
+	const { chain: network } = useNetwork();
+	const futuresEndpoint = getFuturesEndpoint(network?.id as NetworkId);
+	const { isConnected: isWalletConnected } = useAccount();
+
+	const isL2 =
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply queries/futures/useGetAllFuturesTradesForAccount.ts:20]: Good catch. Updated.

File: sections/shared/components/GasPriceSelect/GasPriceSelect.tsx (hunk 8 above)
Comment (LeifuChen) [reply sections/shared/components/GasPriceSelect/GasPriceSelect.tsx:28]: This is a combination of the old naming convention. https://github.com/Kwenta/kwenta/blob/dev/store/wallet/index.ts#L25-L28

I tried to combine the following four states into two hooks to check the network id.
`isMainnetState => L1 Mainnet`
`isL1KovanState => L1 Kovan` (never used)
`isL2State => L2 Mainnet` 
`isL2KovanState => L2 Kovan` (never used)

File: sections/exchange/FooterCard/TradeSummaryCard/GasPriceSummaryItem.tsx (hunk 9 above)
Comment (LeifuChen) [reply sections/exchange/FooterCard/TradeSummaryCard/GasPriceSummaryItem.tsx:37]: Yes. This is supposed to use `useIsL1` hook.

**Review by Tburm — DISMISSED** [review]

**Review by platschi — APPROVED** [review]

**Pull request merged**